src/storage.rs

```diff
@@ -36,6 +36,43 @@ impl<I2C, PS, AS, CD> Storage<I2C, PS, AS, CD> {
     pub fn destroy(self) -> (I2C, CD) {
         (self.eeprom.destroy(), self.count_down)
     }
+
+    /// disables polling for the eeprom
+    pub fn disable_polling(mut self) {
+        self.eeprom.polling_active = false;
+    }
+
+    /// enables polling for the current eeprom, if the eeprom supports it
+    pub fn enable_polling(mut self) {
+        if let crate::PollingSupport::Polling = self.eeprom.polling {
+            self.eeprom.polling_active = true;
+        }
+    }
+}
+
+impl<I2C, E, PS, AS, CD> Storage<I2C, PS, AS, CD>
+where
+    I2C: Write<Error = E> + WriteRead<Error = E>,
+    AS: MultiSizeAddr,
+    CD: CountDown<Time = Duration>,
+{
+    fn wait(&mut self) -> Result<(), Error<E>> {
+        loop {
+            match self.count_down.wait() {
+                Err(nb::Error::WouldBlock) => {
+                    if self.eeprom.polling_active {
+                        match self.eeprom.read_byte(0) {
+                            Ok(_) => break Ok(()),   // done
+                            Err(Error::I2C(_)) => {} // not ready, repeat
+                            Err(e) => break Err(e),
```

Comment:
Looking at this again, it seems we should cancel the countdown in both `break` cases here, since it will be started again in the caller function while it might still be running. What happens in that case is unspecified in the interface so it might panic in some implementations.

Reply:
It is not possible to stop a countdown once it was started. But i think that the start function always starts a new countdown and doesn't use the existing one, so this wouldn't be a problem.

Reply:
Hmm, there is another trait for [`Cancel`](https://docs.rs/embedded-hal/latest/embedded_hal/timer/trait.Cancel.html) but you are right that the documentation for `start` says "starts a _new_ countdown" without mention of reuse so we can leave this as-is.